<p style="text-align:center">
    <a href="https://skills.network/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkPY0220ENSkillsNetwork900-2022-01-01" target="_blank">
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/assets/logos/SN_web_lightmode.png" width="200" alt="Skills Network Logo">
    </a>
</p>


<h1>Extracting and Visualizing Stock Data</h1>
<h2>Description</h2>


Extracting essential data from a dataset and displaying it is a necessary part of data science; therefore individuals can make correct decisions based on the data. In this assignment, you will extract some stock data, you will then display this data in a graph.


<h2>Table of Contents</h2>
<div class="alert alert-block alert-info" style="margin-top: 20px">
    <ul>
        <li>Define a Function that Makes a Graph</li>
        <li>Question 1: Use yfinance to Extract Stock Data</li>
        <li>Question 2: Use Webscraping to Extract Tesla Revenue Data</li>
        <li>Question 3: Use yfinance to Extract Stock Data</li>
        <li>Question 4: Use Webscraping to Extract GME Revenue Data</li>
        <li>Question 5: Plot Tesla Stock Graph</li>
        <li>Question 6: Plot GameStop Stock Graph</li>
    </ul>
<p>
    Estimated Time Needed: <strong>30 min</strong></p>
</div>

<hr>


***Note***:- If you are working Locally using anaconda, please uncomment the following code and execute it.
Use the version as per your python version.


In [1]:
!pip install yfinance
!pip install bs4
!pip install nbformat
!pip install --upgrade plotly

   ---------------------------------------- 0.0/14.8 MB ? eta -:--:--
   ----------------- ---------------------- 6.6/14.8 MB 33.5 MB/s eta 0:00:01
   ------------------------------------ --- 13.6/14.8 MB 34.2 MB/s eta 0:00:01
   ---------------------------------------- 14.8/14.8 MB 29.1 MB/s eta 0:00:00
  Attempting uninstall: plotly
    Found existing installation: plotly 5.24.1
    Uninstalling plotly-5.24.1:
      Successfully uninstalled plotly-5.24.1


In [2]:
import yfinance as yf
import pandas as pd
import requests
from bs4 import BeautifulSoup
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [3]:
import plotly.io as pio
pio.renderers.default = "iframe"

In Python, you can ignore warnings using the warnings module. You can use the filterwarnings function to filter or ignore specific warning messages or categories.


In [4]:
import warnings
# Ignore all warnings
warnings.filterwarnings("ignore", category=FutureWarning)

## Define Graphing Function


In this section, we define the function `make_graph`. **You don't have to know how the function works, you should only care about the inputs. It takes a dataframe with stock data (dataframe must contain Date and Close columns), a dataframe with revenue data (dataframe must contain Date and Revenue columns), and the name of the stock.**


In [5]:
def make_graph(stock_data, revenue_data, stock):
    fig = make_subplots(rows=2, cols=1, shared_xaxes=True, subplot_titles=("Historical Share Price", "Historical Revenue"), vertical_spacing = .3)
    stock_data_specific = stock_data[stock_data.Date <= '2021-06-14']
    revenue_data_specific = revenue_data[revenue_data.Date <= '2021-04-30']
    fig.add_trace(go.Scatter(x=pd.to_datetime(stock_data_specific.Date, infer_datetime_format=True), y=stock_data_specific.Close.astype("float"), name="Share Price"), row=1, col=1)
    fig.add_trace(go.Scatter(x=pd.to_datetime(revenue_data_specific.Date, infer_datetime_format=True), y=revenue_data_specific.Revenue.astype("float"), name="Revenue"), row=2, col=1)
    fig.update_xaxes(title_text="Date", row=1, col=1)
    fig.update_xaxes(title_text="Date", row=2, col=1)
    fig.update_yaxes(title_text="Price ($US)", row=1, col=1)
    fig.update_yaxes(title_text="Revenue ($US Millions)", row=2, col=1)
    fig.update_layout(showlegend=False,
    height=900,
    title=stock,
    xaxis_rangeslider_visible=True)
    fig.show()
    from IPython.display import display, HTML
    fig_html = fig.to_html()
    display(HTML(fig_html))

Use the make_graph function that we’ve already defined. You’ll need to invoke it in questions 5 and 6 to display the graphs and create the dashboard. 
> **Note: You don’t need to redefine the function for plotting graphs anywhere else in this notebook; just use the existing function.**


## Question 1: Use yfinance to Extract Stock Data


Using the `Ticker` function enter the ticker symbol of the stock we want to extract data on to create a ticker object. The stock is Tesla and its ticker symbol is `TSLA`.


In [13]:
def make_graph(tesla_data, revenue_data, tesla):
    fig = make_subplots(rows=2, cols=1, shared_xaxes=True, subplot_titles=("Historical Share Price", "Historical Revenue"), vertical_spacing = .3)
    tesla_data_specific = tesla_data[tesla_data.Date <= '2021-06-14']
    revenue_data_specific = revenue_data[revenue_data.Date <= '2021-04-30']
    fig.add_trace(go.Scatter(x=pd.to_datetime(tesla_data_specific.Date, inplace=True), y=tesla_data_specific.Close.astype("float"), name="Share Price"), row=1, col=1)
    fig.add_trace(go.Scatter(x=pd.to_datetime(revenue_data_specific.Date, inplace=True), y=revenue_data_specific.Revenue.astype("float"), name="Revenue"), row=2, col=1)
    fig.update_xaxes(title_text="Date", row=1, col=1)
    fig.update_xaxes(title_text="Date", row=2, col=1)
    fig.update_yaxes(title_text="Price ($US)", row=1, col=1)
    fig.update_yaxes(title_text="Revenue ($US Millions)", row=2, col=1)
    fig.update_layout(showlegend=False,
    height=900,
    title=tesla,
    xaxis_rangeslider_visible=True)
    fig.show()
    from IPython.display import display, HTML
    fig_html = fig.to_html()
    display(HTML(fig_html))

In [14]:
tesla_ticker = yf.Ticker("TSLA")

Using the ticker object and the function `history` extract stock information and save it in a dataframe named `tesla_data`. Set the `period` parameter to ` "max" ` so we get information for the maximum amount of time.


In [15]:
tesla_data = tesla_ticker.history(period="max")
tesla_hist = tesla_data

**Reset the index** using the `reset_index(inplace=True)` function on the tesla_data DataFrame and display the first five rows of the `tesla_data` dataframe using the `head` function. Take a screenshot of the results and code from the beginning of Question 1 to the results below.


In [18]:
tesla_hist.head()

Open      High       Low     Close     Volume  \
Date                                                                           
2010-06-29 00:00:00-04:00  1.266667  1.666667  1.169333  1.592667  281494500   
2010-06-30 00:00:00-04:00  1.719333  2.028000  1.553333  1.588667  257806500   
2010-07-01 00:00:00-04:00  1.666667  1.728000  1.351333  1.464000  123282000   
2010-07-02 00:00:00-04:00  1.533333  1.540000  1.247333  1.280000   77097000   
2010-07-06 00:00:00-04:00  1.333333  1.333333  1.055333  1.074000  103003500   

                           Dividends  Stock Splits  
Date                                                
2010-06-29 00:00:00-04:00        0.0           0.0  
2010-06-30 00:00:00-04:00        0.0           0.0  
2010-07-01 00:00:00-04:00        0.0           0.0  
2010-07-02 00:00:00-04:00        0.0           0.0  
2010-07-06 00:00:00-04:00        0.0           0.0

## Question 2: Use Webscraping to Extract Tesla Revenue Data


Use the `requests` library to download the webpage https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/revenue.htm Save the text of the response as a variable named `html_data`.


In [92]:
url = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/revenue.htm"

response = requests.get(url)
html_data = response.content

Parse the html data using `beautiful_soup` using parser i.e `html5lib` or `html.parser`.


In [94]:
soup = BeautifulSoup(html_data, 'html.parser')

Using `BeautifulSoup` or the `read_html` function extract the table with `Tesla Revenue` and store it into a dataframe named `tesla_revenue`. The dataframe should have columns `Date` and `Revenue`.


<details><summary>Step-by-step instructions</summary>

```

Here are the step-by-step instructions:

1. Create an Empty DataFrame
2. Find the Relevant Table
3. Check for the Tesla Quarterly Revenue Table
4. Iterate Through Rows in the Table Body
5. Extract Data from Columns
6. Append Data to the DataFrame

```
</details>


<details><summary>Click here if you need help locating the table</summary>

```
    
Below is the code to isolate the table, you will now need to loop through the rows and columns like in the previous lab
    
soup.find_all("tbody")[1]
    
If you want to use the read_html function the table is located at index 1

We are focusing on quarterly revenue in the lab.
```

</details>


In [ ]:
tesla_revenue = pd.DataFrame(columns=["Date", "Revenue"])
rows = []

for row in soup.find_all("tbody")[1].find_all("tr"):
    col = row.find_all("td")
    date = col[0].text
    revenue = col[1].text.replace("$", "").replace(",", "")
    rows.append({"Date": date, "Revenue": revenue})

tesla_revenue = pd.concat([tesla_revenue, pd.DataFrame(rows)], ignore_index=True)

Extracted rows: [{'Date': '2022-09-30', 'Revenue': '21454'}, {'Date': '2022-06-30', 'Revenue': '16934'}, {'Date': '2022-03-31', 'Revenue': '18756'}, {'Date': '2021-12-31', 'Revenue': '17719'}, {'Date': '2021-09-30', 'Revenue': '13757'}, {'Date': '2021-06-30', 'Revenue': '11958'}, {'Date': '2021-03-31', 'Revenue': '10389'}, {'Date': '2020-12-31', 'Revenue': '10744'}, {'Date': '2020-09-30', 'Revenue': '8771'}, {'Date': '2020-06-30', 'Revenue': '6036'}, {'Date': '2020-03-31', 'Revenue': '5985'}, {'Date': '2019-12-31', 'Revenue': '7384'}, {'Date': '2019-09-30', 'Revenue': '6303'}, {'Date': '2019-06-30', 'Revenue': '6350'}, {'Date': '2019-03-31', 'Revenue': '4541'}, {'Date': '2018-12-31', 'Revenue': '7226'}, {'Date': '2018-09-30', 'Revenue': '6824'}, {'Date': '2018-06-30', 'Revenue': '4002'}, {'Date': '2018-03-31', 'Revenue': '3409'}, {'Date': '2017-12-31', 'Revenue': '3288'}, {'Date': '2017-09-30', 'Revenue': '2985'}, {'Date': '2017-06-30', 'Revenue': '2790'}, {'Date': '2017-03-31', 'Reven

Execute the following line to remove the comma and dollar sign from the `Revenue` column. 


In [166]:
tesla_revenue["Revenue"] = tesla_revenue["Revenue"].astype(str).str.replace(",", "").str.replace("$", "")

Execute the following lines to remove an null or empty strings in the Revenue column.


In [167]:
tesla_revenue.dropna(inplace=True)

tesla_revenue = tesla_revenue[tesla_revenue["Revenue"] != ""]

Display the last 5 row of the `tesla_revenue` dataframe using the `tail` function. Take a screenshot of the results.


In [169]:
tesla_revenue.tail()

Date Revenue
48  2010-09-30      31
49  2010-06-30      28
50  2010-03-31      21
52  2009-09-30      46
53  2009-06-30      27

## Question 3: Use yfinance to Extract Stock Data


Using the `Ticker` function enter the ticker symbol of the stock we want to extract data on to create a ticker object. The stock is GameStop and its ticker symbol is `GME`.


In [173]:
gme_ticker = yf.Ticker("GME")

Using the ticker object and the function `history` extract stock information and save it in a dataframe named `gme_data`. Set the `period` parameter to ` "max" ` so we get information for the maximum amount of time.


In [174]:
gme_data = gme_ticker.history(period="max")
gme_hist = gme_data

**Reset the index** using the `reset_index(inplace=True)` function on the gme_data DataFrame and display the first five rows of the `gme_data` dataframe using the `head` function. Take a screenshot of the results and code from the beginning of Question 3 to the results below.


In [175]:
gme_data.head()

Open      High       Low     Close    Volume  \
Date                                                                          
2002-02-13 00:00:00-05:00  1.620128  1.693350  1.603296  1.691667  76216000   
2002-02-14 00:00:00-05:00  1.712707  1.716074  1.670626  1.683250  11021600   
2002-02-15 00:00:00-05:00  1.683250  1.687458  1.658002  1.674834   8389600   
2002-02-19 00:00:00-05:00  1.666418  1.666418  1.578047  1.607504   7410400   
2002-02-20 00:00:00-05:00  1.615920  1.662210  1.603296  1.662210   6892800   

                           Dividends  Stock Splits  
Date                                                
2002-02-13 00:00:00-05:00        0.0           0.0  
2002-02-14 00:00:00-05:00        0.0           0.0  
2002-02-15 00:00:00-05:00        0.0           0.0  
2002-02-19 00:00:00-05:00        0.0           0.0  
2002-02-20 00:00:00-05:00        0.0           0.0

## Question 4: Use Webscraping to Extract GME Revenue Data


Use the `requests` library to download the webpage https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/stock.html. Save the text of the response as a variable named `html_data_2`.


In [176]:
url = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/stock.html"

response = requests.get(url)
html_data_2 = response.content

Parse the html data using `beautiful_soup` using parser i.e `html5lib` or `html.parser`.


In [177]:
soup = BeautifulSoup(html_data_2, 'html.parser') 

Using `BeautifulSoup` or the `read_html` function extract the table with `GameStop Revenue` and store it into a dataframe named `gme_revenue`. The dataframe should have columns `Date` and `Revenue`. Make sure the comma and dollar sign is removed from the `Revenue` column.


> **Note: Use the method similar to what you did in question 2.**  


<details><summary>Click here if you need help locating the table</summary>

```
    
Below is the code to isolate the table, you will now need to loop through the rows and columns like in the previous lab
    
soup.find_all("tbody")[1]
    
If you want to use the read_html function the table is located at index 1


```

</details>


In [178]:
gme_revenue = pd.DataFrame(columns=["Date", "Revenue"])
rows = []

for row in soup.find_all("tbody")[1].find_all("tr"):
    col = row.find_all("td")
    date = col[0].text
    revenue = col[1].text.replace("$", "").replace(",", "")
    rows.append({"Date": date, "Revenue": revenue})

print("Extracted rows:", rows)

gme_revenue = pd.concat([gme_revenue, pd.DataFrame(rows)], ignore_index=True)

Extracted rows: [{'Date': '2020-04-30', 'Revenue': '1021'}, {'Date': '2020-01-31', 'Revenue': '2194'}, {'Date': '2019-10-31', 'Revenue': '1439'}, {'Date': '2019-07-31', 'Revenue': '1286'}, {'Date': '2019-04-30', 'Revenue': '1548'}, {'Date': '2019-01-31', 'Revenue': '3063'}, {'Date': '2018-10-31', 'Revenue': '1935'}, {'Date': '2018-07-31', 'Revenue': '1501'}, {'Date': '2018-04-30', 'Revenue': '1786'}, {'Date': '2018-01-31', 'Revenue': '2825'}, {'Date': '2017-10-31', 'Revenue': '1989'}, {'Date': '2017-07-31', 'Revenue': '1688'}, {'Date': '2017-04-30', 'Revenue': '2046'}, {'Date': '2017-01-31', 'Revenue': '2403'}, {'Date': '2016-10-31', 'Revenue': '1959'}, {'Date': '2016-07-31', 'Revenue': '1632'}, {'Date': '2016-04-30', 'Revenue': '1972'}, {'Date': '2016-01-31', 'Revenue': '3525'}, {'Date': '2015-10-31', 'Revenue': '2016'}, {'Date': '2015-07-31', 'Revenue': '1762'}, {'Date': '2015-04-30', 'Revenue': '2061'}, {'Date': '2015-01-31', 'Revenue': '3476'}, {'Date': '2014-10-31', 'Revenue': '20

Display the last five rows of the `gme_revenue` dataframe using the `tail` function. Take a screenshot of the results.


In [179]:
gme_revenue.tail()

Date Revenue
57  2006-01-31    1667
58  2005-10-31     534
59  2005-07-31     416
60  2005-04-30     475
61  2005-01-31     709

## Question 5: Plot Tesla Stock Graph


Use the `make_graph` function to graph the Tesla Stock Data, also provide a title for the graph. Note the graph will only show data upto June 2021.


<details><summary>Hint</summary>

```

You just need to invoke the make_graph function with the required parameter to print the graphs.The structure to call the `make_graph` function is `make_graph(tesla_data, tesla_revenue, 'Tesla')`.

```
    
</details>


In [3]:
import pandas as pd
import plotly.graph_objects as go
from plotly.subplots import make_subplots
from bs4 import BeautifulSoup
import requests
import yfinance as yf

url = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/revenue.htm"
response = requests.get(url)
html_data = response.content

soup = BeautifulSoup(html_data, "html.parser")

rows = []

for row in soup.find("tbody").find_all("tr"):
    col = row.find_all("td")
    if len(col) >= 2: 
        date = col[0].text.strip()
        revenue = col[1].text.strip().replace("$", "").replace(",", "")
        rows.append({"Date": date, "Revenue": revenue})

tesla_revenue = pd.DataFrame(rows)

tesla_revenue["Date"] = pd.to_datetime(tesla_revenue["Date"], errors="coerce")

tesla_revenue["Revenue"] = pd.to_numeric(tesla_revenue["Revenue"], errors='coerce')

tesla_revenue.dropna(inplace=True)

tesla_data = yf.Ticker("TSLA").history(period="max")
tesla_data.reset_index(inplace=True)

tesla_data["Date"] = tesla_data["Date"].dt.tz_localize(None)

print("tesla_revenue missing values:\n", tesla_revenue.isnull().sum())
print("tesla_data missing values:\n", tesla_data.isnull().sum())

print("tesla_data columns:", tesla_data.columns)
print("tesla_revenue columns:", tesla_revenue.columns)
print("tesla_data head:\n", tesla_data.head())
print("tesla_revenue head:\n", tesla_revenue.head())

def make_graph(tesla_data, tesla_revenue, title):
    fig = make_subplots(rows=2, cols=1, shared_xaxes=True, subplot_titles=("Historical Share Price", "Historical Revenue"), vertical_spacing=0.3)

    tesla_data_specific = tesla_data[tesla_data['Date'] <= pd.to_datetime('2021-06-14')]
    tesla_revenue_specific = tesla_revenue[tesla_revenue['Date'] <= pd.to_datetime('2021-04-30')]

    fig.add_trace(go.Scatter(x=tesla_data_specific['Date'], y=tesla_data_specific['Close'], name="Share Price"), row=1, col=1)
    fig.add_trace(go.Scatter(x=tesla_revenue_specific['Date'], y=tesla_revenue_specific['Revenue'], name="Revenue"), row=2, col=1)

    fig.update_xaxes(title_text="Date", row=1, col=1)
    fig.update_xaxes(title_text="Date", row=2, col=1)
    fig.update_yaxes(title_text="Price ($US)", row=1, col=1)
    fig.update_yaxes(title_text="Revenue ($US Millions)", row=2, col=1)

    fig.update_layout(
        showlegend=False,
        height=900,
        title=title,
        xaxis_rangeslider_visible=True
    )

    fig.show()

make_graph(tesla_data, tesla_revenue, 'Tesla Stock Data')

tesla_revenue missing values:
 Date       0
Revenue    0
dtype: int64
tesla_data missing values:
 Date            0
Open            0
High            0
Low             0
Close           0
Volume          0
Dividends       0
Stock Splits    0
dtype: int64
tesla_data columns: Index(['Date', 'Open', 'High', 'Low', 'Close', 'Volume', 'Dividends',
       'Stock Splits'],
      dtype='object')
tesla_revenue columns: Index(['Date', 'Revenue'], dtype='object')
tesla_data head:
         Date      Open      High       Low     Close     Volume  Dividends  \
0 2010-06-29  1.266667  1.666667  1.169333  1.592667  281494500        0.0   
1 2010-06-30  1.719333  2.028000  1.553333  1.588667  257806500        0.0   
2 2010-07-01  1.666667  1.728000  1.351333  1.464000  123282000        0.0   
3 2010-07-02  1.533333  1.540000  1.247333  1.280000   77097000        0.0   
4 2010-07-06  1.333333  1.333333  1.055333  1.074000  103003500        0.0   

   Stock Splits  
0           0.0  
1           0.0  
2 

## Question 6: Plot GameStop Stock Graph


Use the `make_graph` function to graph the GameStop Stock Data, also provide a title for the graph. The structure to call the `make_graph` function is `make_graph(gme_data, gme_revenue, 'GameStop')`. Note the graph will only show data upto June 2021.


<details><summary>Hint</summary>

```

You just need to invoke the make_graph function with the required parameter to print the graphs.The structure to call the `make_graph` function is `make_graph(gme_data, gme_revenue, 'GameStop')`

```
    
</details>


In [4]:
import pandas as pd
import plotly.graph_objects as go
from plotly.subplots import make_subplots
from bs4 import BeautifulSoup
import requests
import yfinance as yf

url = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0220EN-SkillsNetwork/labs/project/stock.html"
response = requests.get(url)
html_data = response.content

soup = BeautifulSoup(html_data, "html.parser")

gme_revenue = pd.DataFrame(columns=["Date", "Revenue"])

rows = []

for row in soup.find("tbody").find_all("tr"):
    col = row.find_all("td")
    if len(col) >= 2:
        date = col[0].text.strip()
        revenue = col[1].text.strip().replace("$", "").replace(",", "")
        rows.append({"Date": date, "Revenue": revenue})

gme_revenue = pd.DataFrame(rows)

gme_revenue["Date"] = pd.to_datetime(gme_revenue["Date"], errors="coerce")

gme_revenue["Revenue"] = pd.to_numeric(gme_revenue["Revenue"], errors="coerce")

gme_revenue.dropna(inplace=True)

gme_data = yf.Ticker("GME").history(period="max")
gme_data.reset_index(inplace=True)

gme_data["Date"] = gme_data["Date"].dt.tz_localize(None)

print("gme_data columns:", gme_data.columns)
print("gme_revenue columns:", gme_revenue.columns)
print("gme_data head:", gme_data.head())
print("gme_revenue head:", gme_revenue.head())

def make_graph(gme_data, gme_revenue, title):
    fig = make_subplots(rows=2, cols=1, shared_xaxes=True, subplot_titles=("Historical Share Price", "Historical Revenue"), vertical_spacing=0.3)

    gme_data_specific = gme_data[gme_data['Date'] <= pd.to_datetime('2021-06-14')]
    gme_revenue_specific = gme_revenue[gme_revenue['Date'] <= pd.to_datetime('2021-04-30')]

    fig.add_trace(go.Scatter(x=gme_data_specific['Date'], y=gme_data_specific['Close'], name="Share Price"), row=1, col=1)
    fig.add_trace(go.Scatter(x=gme_revenue_specific['Date'], y=gme_revenue_specific['Revenue'], name="Revenue"), row=2, col=1)

    fig.update_xaxes(title_text="Date", row=1, col=1)
    fig.update_xaxes(title_text="Date", row=2, col=1)
    fig.update_yaxes(title_text="Price ($US)", row=1, col=1)
    fig.update_yaxes(title_text="Revenue ($US Millions)", row=2, col=1)

    fig.update_layout(
        showlegend=False,
        height=900,
        title=title,
        xaxis_rangeslider_visible=True
    )

    fig.show()

make_graph(gme_data, gme_revenue, 'GME Stock Data')

gme_data columns: Index(['Date', 'Open', 'High', 'Low', 'Close', 'Volume', 'Dividends',
       'Stock Splits'],
      dtype='object')
gme_revenue columns: Index(['Date', 'Revenue'], dtype='object')
gme_data head:         Date      Open      High       Low     Close    Volume  Dividends  \
0 2002-02-13  1.620128  1.693350  1.603296  1.691667  76216000        0.0   
1 2002-02-14  1.712707  1.716073  1.670626  1.683250  11021600        0.0   
2 2002-02-15  1.683250  1.687458  1.658002  1.674834   8389600        0.0   
3 2002-02-19  1.666417  1.666417  1.578047  1.607504   7410400        0.0   
4 2002-02-20  1.615920  1.662210  1.603296  1.662210   6892800        0.0   

   Stock Splits  
0           0.0  
1           0.0  
2           0.0  
3           0.0  
4           0.0  
gme_revenue head:         Date  Revenue
0 2020-01-01     6466
1 2019-01-01     8285
2 2018-01-01     8547
3 2017-01-01     7965
4 2016-01-01     9364


<h2>About the Authors:</h2> 

<a href="https://www.linkedin.com/in/joseph-s-50398b136/">Joseph Santarcangelo</a> has a PhD in Electrical Engineering, his research focused on using machine learning, signal processing, and computer vision to determine how videos impact human cognition. Joseph has been working for IBM since he completed his PhD.

Azim Hirjani


## Change Log

| Date (YYYY-MM-DD) | Version | Changed By    | Change Description        |
| ----------------- | ------- | ------------- | ------------------------- |
| 2022-02-28        | 1.2     | Lakshmi Holla | Changed the URL of GameStop |
| 2020-11-10        | 1.1     | Malika Singla | Deleted the Optional part |
| 2020-08-27        | 1.0     | Malika Singla | Added lab to GitLab       |

<hr>

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>

<p>
